platform-dom/src/main/java/com/softicar/platform/dom/elements/popup/configuration/DomPopupChildClosingMode.java

```diff
@@ -0,0 +1,68 @@
+package com.softicar.platform.dom.elements.popup.configuration;
+
+import com.softicar.platform.dom.elements.popup.DomPopup;
+import com.softicar.platform.dom.node.IDomNode;
+
+/**
+ * Enumerates possible child-popup-closing modes of a {@link DomPopup}.
+ * <p>
+ * Definition: There is a parent-child relation between two {@link DomPopup}
+ * elements P1 and P2 if, and only if, P2 was opened from an {@link IDomNode} in
+ * P1.
+ *
+ * @author Alexander Schmidt
+ */
+public enum DomPopupChildClosingMode {
```

Comment:
Generally, I tried hard to pick short names (probably too hard) because the primary way of selecting a mode would be a convenience setter on `DomPopupConfiguration`, e.g. `setChildClosingModeAutomaticAll`.
If we indeed rename those enumerators to something longer, I think the convenience methods on `DomPopupConfiguration` need to be removed. At least, I think that `setChildClosingModeNoConfirmationAndCloseChildren` is no longer comprehensible.
That being said, each enumerator in `DomPopupDisplayMode` defines a sane default `DomPopupChildClosingMode`... deviations/customizations will probably be extremely rare; so related convenience might be irrelevant anyway.

> which modes are in use at all at the moment?

- Currently, `AUTOMATIC_NONE` and `INTERACTIVE_ALL` are in use.
- Frontend devs once mentioned that something like `INTERACTIVE_OPTIONAL` would be useful.
- `AUTOMATIC_ALL` could be useful if an application tries to control popups. I originally thought that this was required for one of the display modes, which turned out wrong... if I were to discard a mode, then I'd pick this one.


> two distinct enumerations

I considered this as well - the issue here is, that there's a lack of symmetry in behaviors. `AUTOMATIC_ALL` and `INTERACTIVE_ALL` correspond to each other; but `AUTOMATIC_NONE` and `INTERACTIVE_OPTIONAL` don't.

All in all, I'm not sure what to do now.

Reply:
Somehow this reminds me of the complexity around auto-completes, where we had support for too many cornercases which made it impossible to understand anything. We already have a set of display modes for popups, and here we have a list of intertwined closing modes for popups and their children. This probably spans a tree of uncountable number of possible configurations.

Reply:
I suggest to cut this down to simple set of 2 or 3 modes overall. Maybe create a story, or do it in a follow up PR?

Reply:
- I think this is nowhere near the complexity of the auto-complete input code.
- It's tempting to simplify this right now, and to tightly couple child-closing modes to display modes - but it _might_ be a mistake (as in 'removing required flexibility').
- Created PLAT-863 to address this.